applications/FluidDynamicsApplication/custom_processes/embedded_skin_visualization_process.cpp

```diff
@@ -89,6 +90,17 @@ ModelPart& EmbeddedSkinVisualizationProcess::CreateAndPrepareVisualizationModelP
         r_visualization_variables_list.Add(r_var);
     }
 
+    // Set the origin model part as temporary communicator
+    // This is required to perform all the IsDistributed checks that appear before creating the visualization mesh
+    // This will be updated by a proper one after the creation of the visualization entities
+    r_visualization_model_part.SetCommunicator(r_origin_model_part.pGetCommunicator());
+
+    // If MPI, add the PARTITION_INDEX variable to the visualization model part variables
+    // Note that check the IsDistributed with the default communicator as the visualization model part one is not set yet
+    if (ParallelEnvironment::GetDefaultDataCommunicator().IsDistributed()) {
```

Comment:
don't use this, this must come from the ModelPart!

Reply:
I can't. As the comment explains the `ModelPart` communicator is not set so the inner pointer points to the default serial one. 
Actually, I struggle with this for a while so I'm pretty convinced in this case :smile:

Reply:
Actually I think that the `SetCommunicator` above is not required...

Reply:
why is it not set yet?
This then seems like a bigger design issue to me
can you point me to the place where you call this ?

Reply:
(forget about the previous comment; the communicator needs to be set to perform the `IsDistributed` checks outside this process)

It is not set because the visualization model part is created in here and cannot be created at the solver level. As the visualization mesh contains different entities than that ones of the parent model part, the communicator should be different. However the communication plan cannot be established until we create such new entities (even though there are no ghost nodes so far). Until these new entities are created we need a "fake" communicator in order to do the `IsDistributed` checks (these are for instance required in order to create the output processes of the visualization mesh).